docs/_mitigations/mi-18_agent-authority-least-privilege-framework.md

```diff
@@ -0,0 +1,159 @@
+* **Separation of Duties Enforcement**: Multi-step processes requiring approval or verification should be enforced through agent privilege restrictions, preventing single agents from completing entire high-risk workflows.
+* **Dynamic Privilege De-escalation**: Agent privileges should automatically reduce to minimum levels when not actively engaged in authorized tasks.
+* **Business Logic Enforcement**: Access controls should enforce business rules, approval limits, and regulatory requirements at the privilege level, not just through application logic.
+
+---
+
+## Implementation Guidance
+
+### 1. Agent Role and Privilege Definition
+
+* **Role-Based Agent Classification**:
+  * **Customer Service Agents**: Read-only access to customer account information, limited transaction inquiry capabilities, no modification or transfer authorities.
+  * **Risk Assessment Agents**: Access to risk calculation APIs and customer financial data for analysis purposes only, no decision execution capabilities.
+  * **Compliance Agents**: Read-only access to transaction data and regulatory databases for compliance checking, no approval or modification authorities.
+  * **Trading Agents**: Limited access to market data APIs and position management systems within defined risk parameters and position limits.
+  * **Document Processing Agents**: Access to document analysis APIs and storage systems, no customer-facing or decision-making capabilities.
+
+* **Privilege Matrices and Documentation**:
+  * Maintain comprehensive matrices documenting exactly which APIs, endpoints, and data sources each agent type can access.
+  * Document the business justification for each privilege grant and regularly review privilege assignments.
+  * Implement version control for privilege definitions to track changes over time.
+
+### 2. Dynamic Privilege Management
+
+* **Context-Aware Access Controls**:
+  * **Transaction Value Thresholds**: Restrict agent access to high-value transaction APIs based on configurable monetary limits.
+  * **Customer Sensitivity Levels**: Implement additional access restrictions for VIP customers, high-net-worth individuals, or customers with privacy flags.
+  * **Time-Based Restrictions**: Limit agent access to certain APIs during off-hours, weekends, or maintenance windows.
+  * **Geographic Restrictions**: Restrict agent access based on customer location, regulatory jurisdiction, or data residency requirements.
+
+* **Privilege Escalation Controls**:
+  * Implement controlled privilege escalation mechanisms requiring human approval for agents needing temporary additional access.
+  * Log and monitor all privilege escalation requests and approvals.
+  * Automatic privilege de-escalation after specified time periods or task completion.
+
+### 3. API and Tool Access Enforcement
+
+* **Tool Manager Security Layer**:
+  * Implement comprehensive authorization checks at the tool manager level before any API calls are executed.
```

Comment:
as per my other comment, I think "tool manager" needs to be defined - although it isn't a term I've come across before